carbon-design-system/carbon
# Fixed the startDate to endDate on start input

Closes #16734

After the upgrade from [4.6.9 to 4.6.13](https://github.com/carbon-design-system/carbon/pull/14993) in flatpickr by renovate we started to have a couple new issues in our `DatePicker` component. 
I'm reverting the changes from this [PR](https://github.com/carbon-design-system/carbon/pull/16202/files#:~:text=%5Bi%5D-,input%20%3D%20!dates%5Bi%5D,-%3F%20%27%27) and fixing the issue by validating the `dates.length`, so the dates cannot be changed unless the user clicks or writes a date.

@riddhybansal I would love your feedback in this one, to make sure I didn't miss anything. The test you wrote is passing, so must be fine.

#### Changelog

**Changed**

- Added a new validation inside the if clause

#### Testing / Reviewing

[StackBlitz](https://stackblitz.com/edit/github-7ujtwx?file=src%2FApp.jsx) example that I copied to use in the storybook

- Make sure the start input doesn't show the `startDate to endDate`.
- Make sure the previous issue that adds the `xx/01/2001` doesn't happen anymore
